### PR TITLE
INS-18246 add opensearch plugin parameter in bundleOptions and oidcProvider in clusterRequest

### PR DESCRIPTION
**Pull request by iclonghoang**
Internal reference - [INS-18246](https://instaclustr.atlassian.net/browse/INS-18246)

The following parameters are added:

- index_management_plugin: in bundleOptions to enable Index Management plugin for ODFE/OpenSearch
- oidc_provider: in clusterRequest to specify OIDC provider used for ODFE Kibana or OpenSearch Dashboards

### CODE REVIEW

File: docs/resources/cluster.md [hunk 1]
```diff
@@ -82,6 +83,7 @@ Property | Description | For Bundles | Default
 `kibana_node_size`|Desired master node size. See [here](https://developer.instaclustr.com/#operation/extendedProvisionRequestHandler) for more details. |Elasticsearch|
 `opensearch_dashboards_node_size`|Desired OpenSearch Dashboards node size. See [here](https://developer.instaclustr.com/#operation/extendedProvisionRequestHandler) for more details. |OpenSearch|
 `security_plugin`|Accepts true/false. Enables Security Plugin. This option gives an extra layer of security to the cluster. This will automatically enable internode encryption. Enforced to be true for all OpenSearch clusters. |Elasticsearch, OpenSearch|false
+`index_management_plugin`|Accepts true/false. Enables Index Management Plugin which allows automating index management activities. |Elasticsearch, OpenSearch|false
```
Comment (arjunrajinstaclustr) [inline docs/resources/cluster.md:86]: Doesn't look like this is added to the schema?

File: Makefile [hunk 2]
```diff
@@ -5,7 +5,7 @@ BIN_NAME=terraform-provider-instaclustr
 # for VERSION, don't add prefix "v", e.g., use "1.9.8" instead of "v1.9.8" as it could break circleCI stuff
 
 
-VERSION=1.17.0
+VERSION=1.17.1
```
Comment (arjunrajinstaclustr) [inline Makefile:8]: Since this is a new feature, should be a minor version bump not patch

**Review by arjunrajinstaclustr — CHANGES_REQUESTED** [review]

File: docs/resources/cluster.md (hunk 1 above)
Comment (iclonghoang) [reply docs/resources/cluster.md:86]: yes, I just added `index_management_plugin` and also `security_plugin` to schema

File: Makefile (hunk 2 above)
Comment (iclonghoang) [reply Makefile:8]: yes, fixed

**Review by arjunrajinstaclustr — DISMISSED** [review]

File: acc_test/data/valid_opensearch_cluster_create.tf [hunk 3]
```diff
@@ -22,11 +22,12 @@ resource "instaclustr_cluster" "validOpenSearch" {
...
     options = {
-      dedicated_master_nodes = false,
-      master_node_size       = "t3.small-v2",
-      data_node_size         = "t3.small-v2",
-      security_plugin        = true,
-      client_encryption      = true
+      dedicated_master_nodes  = false,
+      master_node_size        = "t3.small-v2",
+      data_node_size          = "t3.small-v2",
+      security_plugin         = true,
```
Comment (Cai-Chen) [inline acc_test/data/valid_opensearch_cluster_create.tf:28]: Looking good to me. I think security plugin is always enabled for opensearch. As its not related to this ticket, probably we can update that later.

**Review by Cai-Chen — APPROVED** [review]

File: acc_test/data/valid_opensearch_cluster_create.tf (hunk 3 above)
Comment (iclonghoang) [reply acc_test/data/valid_opensearch_cluster_create.tf:28]: thanks Cai,
security_plugin for opensearch only takes 'true' as value like you said, however, it's still considered as an input parameter.

**Review by arjunrajinstaclustr — DISMISSED** [review]

File: acc_test/resource_redis_test.go [hunk 4]
```diff
@@ -44,7 +44,6 @@ func TestRedisResource(t *testing.T) {
 					testCheckResourceValid(resourceName),
 					testCheckResourceCreated(resourceName, hostname, username, apiKey),
 					checkClusterRunning(resourceName, hostname, username, apiKey),
-					testCheckContactIPCorrect(resourceName, hostname, username, apiKey, 5, 5),
```
Comment (iclonghoang) [inline acc_test/resource_redis_test.go:47]: this line tests whether the cluster is allocated across 5 AZs
Alwyn said this is a test for Cassandra but not needed for Redis, after discussing with ChrisC

**Review by Cai-Chen — APPROVED** [review]

**Review by arjunrajinstaclustr — APPROVED** [review]

**Pull request merged**